### DISCUSSION TITLE
Replace Burning Man with Burning Woman

### DISCUSSION THREAD

**Discussion by huey735** [bisq-network/bisq#5418] [Ideas]
**Summary**
**Burning Man** has custody of the BTC trading fees and finds themselves having to spend a great part of it with mining fees due to the high number of UTXOs they are working with. This proposal suggests a new role - **Burning Woman** - that could be executed by any BSQ holder and would involve the interacting with every _make offer_ and _take offer_ transaction.

# 

BSQ was developed as a way to help redirect the revenue generated by Bisq in a way that there would be no need for an entity holding custody of it and having the role of redistributing it. It has failed in that job as of today because we still have that entity in the role of [Burning Man](#burningman).

#### It puts the custody of the BTC trading fees on the hand of a single entity

> ## At present every trader needs to trust the holder of the Bisq donation address:
> 
> ### Why?
> 
> With the current trade protocol, like the previous one, every btc trading fee goes directly to that address at the make and take offer stage. But also the security deposit and btc trade amount may go there if the amount locked in the 2-of-2 multisig isn't released in a amicable manner by the two traders. After 10 days for altcoins and 20 for national currencies trades, either of the traders can request arbitration and in doing so send the funds to the Donation Address and losing custody of their funds. I'm not sure this is clear to traders.

[@huey735 - continue to read...](https://github.com/bisq-network/proposals/issues/183#issuecomment-597092142)

#### It creates a great number of small UTXOs that are later expensive to spend
When traders pay their trading fee in BSQ, they create a transaction paying the trading fees by burning their BSQ. The sum of BSQ in the inputs of the transaction are bigger than the single output they get back. The difference is the trading fee and is used as mining fee.

When traders pay their trading fee in BTC, they create a transaction sending the trading fees to the [Burning Man](#burningman)'s address.

This method generates a great amount of small UTXOs that then need to be spent by the Burning Man to buy BSQ. This becomes really expensive really quick.

### Solution

The new and improved **Burning Man**, let's call them **Burning Woman**, would participate in the making and taking offers of other traders and get trading fee directly. Like a maker in JoinMarket.

This role would need to remain online at all times in order to participate in all making and taking of Bisq offers. They would participate in multiparty transactions with the trader. They'd provide BSQ inputs and burn BSQ in the transaction and receive the equivalent in BTC in an output of their own.

#### Multiparty transactions as a solution for everything

- Trading BSQ/BTC
    Being that BSQ is colored BTC the trading of the two assets can be done in a single transaction.
    The BTC buyer (Burning Woman) needs to provide a BSQ input.
    The BTC seller needs to provide a BTC input - and maybe a BSQ input if they already have BSQ. This would avoid the creation of unnecessary BSQ UTXOs for the BTC seller.

For example, a trader has 1 Msats and wants to buy BSQ.

1 BSQ = 1000 sats
B = BTC buyer
S = BTC seller

|     | inputs | outputs |
| --- | --- | --- |
| 0   | B - 1000 BSQ | B - 500 BSQ |
| 1   | S - 1 000 000 sats | B - 500 000 sats |
| 2   |  | S - 500 BSQ |
| 3   |     | S - 500 000 sats |

This should be possible with the current implementation of BSQ. We'd just need to make sure to rearrange the order of the outputs so that the first two would be the BSQ outputs.

So we have broken the trading of BSQ/BTC from 5 on-chain transactions to 1 atomic transaction.

What else can we do?

We can apply the same principle to get rid of the Burning Man in a manner where they would no longer have custody of BTC.

It'd work as follow.

Alice is a JAVA developer and this cycle she got 10 000 BSQ. She creates an account on a 24/7 Bisq client that will participate in any making/taking of offers and burn her BSQ in exchange for the maker/taker BTC.

1 BSQ = 1000 sats
A = Alice
T = Trader

|     | inputs | outputs |
| --- | --- | --- |
| 0   | A - 1000 BSQ | A - 995 BSQ |
| 1   | A - 40 000 sats | A - 40 500 sats |
| 2   | T - 10 000 000 sats | T - 1 000 000 sats |
| 3   |     | T - 8 990 000 sats |
|     | mining fees | 10 000 sats |

It would be unrealistic to expect all Bisq contributors would have a 24/7 client. But we don't need to. Most contributors can simply sell their earned BSQ on the market. We'd just need a few Burning Women running 24/7 burning the BSQ.

- What are the incentives to become a Burning Woman?
    They can make profit in the difference between what they charge BSQ sellers and what they get from the trading fees

### Possible issues

* **Connectivity**
The Burning Woman/Women would need to remain online 24/7 in order to fully get rid of the Burning Man. This, I believe, asks way more from the Bisq, BTC and Tor networks than what we currently expect.

* **There needs to be enough confirmed UTXOs online for the taking**
It's important to make sure that the Burning Women have enough BSQ UTXOs in their wallets to service multiple traders at the same time. Or that there are enough Burning Women online at any moment.

* **BitcoinJ can't handle too many UTXOs**
>We do not want to use the BTC change address as the BitcoinJ wallet in Bisq is not suited for receiving a lot of small transactions. The wallet would become very heavy/slow over time (would render it to a full BTC node).

[@chimp1984 - #316](https://github.com/bisq-network/proposals/issues/316#issue-820678359)

Related proposals
[#304 - Make burning man a tradebot and add BSQ auction market ](https://github.com/bisq-network/proposals/issues/304)
[#316 - Distribute trade fee in BTC to DAO stakeholders by using locked up BSQ](https://github.com/bisq-network/proposals/issues/316)
[#317 - Distribute trade fee in BTC by using proof of burn](https://github.com/bisq-network/proposals/issues/317)

# 
<a name="burningman"></a> Burning Man - role responsible for taking the trading fees paid in BTC and use it to buy BSQ and the burn it. This entity has control over the BTC.

**Comment by MwithM** [comment]
This seems a thoughtful and good looking idea, but it doesn't seem to solve the issue with delayed payout tx's. I think that usually, just like now, BM has more funds from [delayed payouts](https://mempool.space/address/34VLFgtFKAtwTdZ5rengTT2g2zC99sWQLC) than from [fees](https://mempool.space/address/38bZBj5peYS3Husdz7AH3gEUiUbYRD951t).
I think that the biggest challenge for Bisq is not how to deal with trading fees paid in BTC (which would be solved by allowing them to be paid only with BSQ) but delayed payouts to DAO the donation address controlled by BM.
For this, I think about two possible solutions: we point the delayed payouts to a [burning address](https://github.com/bisq-network/proposals/issues/135) (an unspendable BTC address) or develop an [Automatic Market Maker](https://github.com/bisq-network/proposals/issues/276): delayed payouts would be sent to an address that would automatically sell the BTC for BSQ and burn them.

**Reply by huey735** [reply]
I think I have this correct:

* [**Burning Man**](https://github.com/bisq-network/roles/issues/80)

  Gathers the trading fees paid in BTC and uses it to buy BSQ from the market at a set price and burns it.

* [**Arbitrator**](https://github.com/bisq-network/roles/issues/93)

  Third party that adjudicates trades that go beyond mediation. They receive the trader's locked funds and pay as they see fit.



You are right, @mwithm, my proposal only replaces the **Burning Man**.

### Forcing the traders to pay for trading fees in BSQ only

It would cause too much friction. I had that concern in mind when I came up with the above proposal. There would always need to be a trade-zero where the new trader would need to acquire BSQ. How would they pay the trading fees there?

Well with the [new mode for trading involving a single transaction](https://github.com/bisq-network/proposals/issues/279) it would make things easier. But I still believe that it would add more friction to the existing list of pain points that traders encounter when trading on Bisq.

My proposal shifts those pain points to the contributors (**Burning Women**) and leaves the traders unaffected by it. They, the traders, get to carry the weight of a bigger transaction because they need to include the inputs and outputs belonging to the **Burning Woman** but it seems like a fair exchange.

### How to deal with the funds that go to arbitration (delayed payout txs)?

These funds should make themselves back to the legitimate owner the fastest way possible. Sometimes traders just forget that they have a pending trade and the other trader sees themselves forced to open arbitration. In those situations it's clear who deserves most of the funds. The process of having the delayed payout tx point to a _Automatic Market Maker_ doesn't make too much sense to me:

* It still would be a third party controlling those funds
* it would increase the difficulty for the trader in the right to get their funds
* it would increase the time necessary for the trader in the right to get their funds - they would need to make an appeal to the DAO and wait a cycle or practically two for it to take effect. And in the end they would end up with BSQ which they would then have to trade for BTC.

### Regarding RSK

I'm quite ignorant about RSK. And believe that it's fair to extend that to most of Bisq's current protocol developer. From the talks we've had, they seem to be most comfortable staying with BTC for the moment. The existence of examples for P2P multi-party transactions tools like:

* [Bip78](https://bips.xyz/78)
  * [BTCPay server](https://docs.btcpayserver.org/Payjoin/)
  * [Onion78](https://github.com/armins88/Onion78)
  * And the growing number of [wallets accepting these standard](https://en.bitcoin.it/wiki/PayJoin_adoption)
* [JoinMarket's P2P market for multi-party transactions](https://github.com/JoinMarket-Org/joinmarket-clientserver)
* [Stowaway](https://docs.samourai.io/en/wallet/privacy-enhanced-transactions) - implemented in JAVA
* [Soroban](https://medium.com/samourai-wallet/wallet-update-0-99-96-introducing-soroban-adc9a36a7ddb)

**Reply by MwithM** [reply]
Burning Man has one function, which is buy and burn BSQ using BTC received from 2 different sources: fees and delayed payouts. Burning Woman would take fees, but not delayed payouts.
Dealing with trading fees it's not a minor issue, and this discussion made me understand better how atomic transactions would work (which would improve a lot BSQ liquidity), but anyway, I wanted to point that this would still leave a big single point of failure, which is dealing with BTC from delayed payouts.

Forcing traders to use BSQ could be achived by using a very high trading fee in BTC or allowing that only to buy BSQ. The best feature BSQ achieves is distributing revenue in a decentralized way and completely removing a SPOF, but this feature it's underused.

> Arbitrator
> Third party that adjudicates trades that go beyond mediation. They receive the trader's locked funds and pay as they see fit.

Arbitrator, as a middleman between DAO and traders, has a right to trade with BM the BSQ from DAO reimbursements with priority while BSQ volume is low, but doesn't receive any locked funds. Once delayed payouts are published, this funds should be considered as destroyed (or donated to the DAO) by the trader who publishes it.

The AMM (Automatic Market Maker) would not remove the arbitrator but the burningman. Traders could still claim to the DAO or the arbitrator to be refunded in BSQ or BTC when funds were sent to the donation address because they disagree with their peer (or is inactive). Funds sent to donation address wouldn't be controlled by an anonymous agent, but a smart contract buying BSQ from the AMM and destroying these BSQ.

**Comment by chimp1984** [comment]
Sorry for the late reply.
The idea is interesting but a not to underestimate risk is that a 2 party protocol would become a 3 party protocol and that comes with higher error rates (network failues,...) and higher complexity.

And as @MwithM said the fee distribution seems the be the smaller problem. The funds from delayed payout txs to the BM are the major issue and as those amounts can be rather high and are not frequent they do not fit into a model to distribute to contributors.
